Examples/SimpleGaussian/SimpleGaussian.py

```diff
@@ -23,27 +23,35 @@
 
 import SimpleITK as sitk
 
-if len(sys.argv) < 4:
-    print("Usage: SimpleGaussian <input> <sigma> <output>")
-    sys.exit(1)
 
-reader = sitk.ImageFileReader()
-reader.SetFileName(sys.argv[1])
-image = reader.Execute()
+def main(args):
+    if len(args) < 4:
+        print("Usage: SimpleGaussian <input> <sigma> <output>")
+        sys.exit(1)
 
-pixelID = image.GetPixelID()
+    reader = sitk.ImageFileReader()
+    reader.SetFileName(args[1])
+    input_image = reader.Execute()
 
-gaussian = sitk.SmoothingRecursiveGaussianImageFilter()
-gaussian.SetSigma(float(sys.argv[2]))
-image = gaussian.Execute(image)
+    pixelID = input_image.GetPixelID()
 
-caster = sitk.CastImageFilter()
-caster.SetOutputPixelType(pixelID)
-image = caster.Execute(image)
+    gaussian = sitk.SmoothingRecursiveGaussianImageFilter()
+    gaussian.SetSigma(float(args[2]))
+    blur_image = gaussian.Execute(input_image)
 
-writer = sitk.ImageFileWriter()
-writer.SetFileName(sys.argv[3])
-writer.Execute(image)
+    caster = sitk.CastImageFilter()
+    caster.SetOutputPixelType(pixelID)
+    blur_image = caster.Execute(blur_image)
 
-if "SITK_NOSHOW" not in os.environ:
-    sitk.Show(image, "Simple Gaussian")
+    writer = sitk.ImageFileWriter()
+    writer.SetFileName(args[3])
+    writer.Execute(blur_image)
+
+    return input_image, blur_image
```

Comment:
What do you think of return a dict instead? `return {"input":input_image, "output":blur_image}`

Reply:
Yeah, that's a bit clearer to read. Examples with multiple input/output images could then have a list like: return {"inputs":[input1, input2], "outputs":[output1, output2]}. At least I think that would work.

Reply:
I would expect each input and output to be a separate entry in the dict.

Reply:
My reasoning for returning a list is the code could just loop through all the inputs (if any) and not worry about how many there were, or worry about parsing names and such. Again, I was hoping to make the code general enough that it wouldn't care about the specifics of different Examples. (two inputs / one output, etc.)

Reply:
Your solution of having each image be a separate entry could have the advantage of allowing the display of the key, possibly in a caption - but I don't know how to get it into Sphinx output. The Python variables aren't available to me in the RST markdown.